### ISSUE TITLE
Drawing exact ahead path in navigation mode

### ISSUE THREAD

**Issue by Dizzy55** [Framstag/libosmscout#1338]
I draw ahead path in navigation mode. For this purpose I have used signal from "NavigationModel". Here is my code

```
    Connections {
        target: Navigation.navigationModel
        onRouteAheadChanged: {
            map.addOverlayObject(0, Navigation.navigationModel.routeWayAhead)
        }
    }
```
But I see strange ways on my screen. Sometimes "routeWayAhead" is longer than needed and my position marker is on it. In this situation user see a tail after marker
![inside](https://user-images.githubusercontent.com/4497358/195708265-1d62bbb1-8d0c-49e3-ab76-c61226ee38f7.png)

Sometimes "routeWayAhead" is shorter than necessary and my position marker is outside of the way
![outside](https://user-images.githubusercontent.com/4497358/195708270-27deb161-8810-4989-8d5c-3451d4c655f3.png)

Is it a way to draw exact path for navigation?

**Comment by Dizzy55** [comment]
Moreover, redrawing the path takes a long time and the marker runs into the path anyway. Does engine have a way to draw ahead path fast like a drawing marker?

**Comment by Framstag** [comment]
@Karry : This sound a problem in the Qt part? Is @Dizzy55 using the code as designed? I assume that the "way ahead"...
- Is the complete way ahead and not clipped to the visible area?
- Is the way in relation to OSM nodes. Especially for the start node. Sounds like the first node must not be the first route node for the rest of the way but the current geo location (at least, if one is "on the way")?

**Comment by Karry** [comment]
Yes, this problem is specific for Qt library. Problem is that the route is rendered by map renderer, and rendering may take several seconds. So the route not refreshed with every position update on purpose...
